### PR TITLE
Add new case write same support on discard disk

### PR DESCRIPTION
**Pull request by qingwangrh**
ID:1980243
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=block_discard_write_same --guestname=RHEL.8.5.0 --driveformat=virtio_scsi,virtio_blk --clone=no

(1/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.block_discard_write_same.local_file: PASS (40.28 s)
 (2/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.block_discard_write_same.scsi_debug.unmap: PASS (36.98 s)
 (3/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.block_discard_write_same.scsi_debug.writesame: PASS (37.89 s)
 (4/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.block_discard_write_same.local_file: PASS (43.83 s)
 (5/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.block_discard_write_same.scsi_debug.unmap: PASS (38.09 s)
 (6/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.block_discard_write_same.scsi_debug.writesame: PASS (36.84 s)

**Comment by qingwangrh** [comment]
@YongxueHong @zhenyzha Please help to review. thanks.

**Comment by zhenyzha** [comment]
git am 2897.patch 
Applying: Add new case write same support on discard disk


```
 (1/8) Host_RHEL.m9.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.arm64-pci: PASS (1431.71 s)
 (2/8) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.arm64-pci: PASS (1421.29 s)
 (3/8) Host_RHEL.m9.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.block_discard_write_same.local_file.arm64-pci: PASS (202.13 s)
 (4/8) Host_RHEL.m9.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.block_discard_write_same.scsi_debug.unmap.arm64-pci: PASS (202.99 s)
 (5/8) Host_RHEL.m9.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.block_discard_write_same.scsi_debug.writesame.arm64-pci: PASS (185.48 s)
 (6/8) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.block_discard_write_same.local_file.arm64-pci: PASS (178.88 s)
 (7/8) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.block_discard_write_same.scsi_debug.unmap.arm64-pci: PASS (185.22 s)
 (8/8) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.block_discard_write_same.scsi_debug.writesame.arm64-pci: PASS (189.46 s)
```
LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Review by zhenyzha — APPROVED** [review]

File: qemu/deps/thin-provision/sg_write_same.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+t1=`tail -n 1 /tmp/t1`
+t2=`tail -n 1 /tmp/t2`
+
+if expr $t2 \> $t1;then
+  echo "Find unexpect result $t1 $t2"
+  exit 1
+fi
+
+echo "Time of sg_write_same $t1:$t2"
```
[the remark this answers is not in the document]
Comment (qingwangrh) [reply qemu/deps/thin-provision/sg_write_same.sh]: updated

**Review by YongxueHong — APPROVED** [review]
LGTM

**Pull request merged**